### ISSUE TITLE
Tls provider compatibility issue with M1 chip

### ISSUE THREAD

**Issue by aquali1968** [terraform-google-modules/terraform-google-vault#150]
### TL;DR

Tls provider compatibility issue with M1 chip

### Expected behavior

success full "terraform init" due to changing tls 2.2.0 to 3.3.0

### Observed behavior

Error: Incompatible provider version
│ 
│ Provider registry.terraform.io/hashicorp/tls v2.2.0 does not have a package available for
│ your current platform, darwin_arm64.
│ 
│ Provider releases are separate from Terraform CLI releases, so not all providers are
│ available for all platforms. Other versions of this provider may have different platforms
│ supported.

### Terraform Configuration

```hcl
module "vault" {
  source         = "terraform-google-modules/vault/google"
  project_id     = var.project_id
  version          = "6.1.1"
  /* region         = var.region
  kms_keyring    = var.kms_keyring
  kms_crypto_key = var.kms_crypto_key */
}
```

### Terraform Version

```sh
Terraform v1.1.9
on darwin_arm64
```

### Additional information

I can see that the same issue was created asking for TLS compatibility on M1, and also can see the fix to 3.3.0, however still getting the tls provider error.
